syntax="proto2";

package apollo.parking.proto;

message ParkingVertices {
  optional double spot_left_top_x = 1;
  optional double spot_left_top_y = 2;
  optional double spot_left_down_x = 3;
  optional double spot_left_down_y = 4;
  optional double spot_right_down_x = 5;
  optional double spot_right_down_y = 6;
  optional double spot_right_top_x = 7;
  optional double spot_right_top_y = 8;
}

message TargetParkingSpotInfo {
optional ParkingVertices target_parking_vertices = 1;
  optional bool parking_inwards = 2;
  //optional bool is_right_side =3;
  optional OtherParkingSpotInfo otherParkingSpotInfo = 3;
}

message OtherParkingSpotInfo {
  optional double obstacle_nums = 1 ;
  repeated ParkingVertices other_parking_vertices = 2;
}